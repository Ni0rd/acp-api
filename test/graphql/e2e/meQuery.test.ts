import './env';
import gql from 'graphql-tag';
import { addGlobalMocks, initApollo } from './helpers';

addGlobalMocks();
const getClient = initApollo();

jest.mock('../../../api/utils/auth', () => {
  return {
    getDecodedTokenFromHeaders: jest.fn().mockReturnValue({
      userId: 1,
    }),
  };
});

jest.mock('../../../api/datasources/OdooUser', () => {
  return jest.fn().mockImplementation(() => {
    return {
      getUserById: jest.fn().mockResolvedValue({
        id: 1,
        firstname: 'John',
        lastname: 'Doe',
        email: 'john.doe@example.net',
      }),
    };
  });
});

describe('me query', () => {
  test('returns the User', async () => {
    const query = {
      query: gql`
        query {
          me {
            id
            email
            firstname
            lastname
          }
        }
      `,
      context: {
        headers: {
          authorization: 'token 123abc',
        },
      },
    };
    const response = await getClient().query(query);
    expect(response).toMatchSnapshot();
  });
});
