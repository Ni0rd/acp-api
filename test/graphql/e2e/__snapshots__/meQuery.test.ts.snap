// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`me query returns the User 1`] = `
Object {
  "data": Object {
    "me": Object {
      "__typename": "User",
      "email": "john.doe@example.net",
      "firstname": "John",
      "id": 1,
      "lastname": "Doe",
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;
